Firmware/cardreader.cpp

```diff
@@ -79,13 +79,13 @@ void CardReader::lsDive(const char *prepend, SdFile parent, const char * const m
 				createFilename(lfilename, p);
 				// Allocate enough stack space for the full path to a folder, trailing slash, and nul
 				bool prepend_is_empty = (prepend[0] == '\0');
-				int len = (prepend_is_empty ? 1 : strlen(prepend)) + strlen(lfilename) + 1 + 1;
+				int len = (prepend_is_empty ? 1 : strlen(prepend)) + strlen(longFilename) + 1 + 1;
```

Comment:
are you sure path[len] won't run out of stack for longer file names?

Reply:
I'm not sure.  I know enough about embedded development to understand your concern, but I don't know enough about this project or AVR development to know how much stack space we have or how close we are to running out.

What I do know is that longFilename can hold 52 characters (plus null terminator) while lfilename can only hold 12 characters.  So there is a potential for an extra 40 characters per level of folder.  It looks like there is a max folder depth of 10 (not as certain about this, just judging variable names).  That would mean there is a potential for an extra 400 bytes of usage at worst case.  I don't know if that is too many extra bytes.

Reply:
Actually it may be worse than the 400 bytes, path is going to be duplicated on the stack at every level of recursion.  The first one would be 40 bytes worse, the next one would be 80 bytes worse, all the way to the last one being 400 bytes worse.  Not sure what the math would really look like there.